sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -707,20 +708,30 @@ object DataSourceStrategy
       agg.aggregateFunction match {
         case aggregate.Min(PushableColumnWithoutNestedColumn(name)) =>
           Some(new Min(FieldReference.column(name)))
+        case aggregate.Min(PushableExpression(expr)) =>
```

Comment:
To simplify the code, can we make `PushableExpression` handle field reference as well? e.g. it returns `FieldReference` if possible, or return `GeneralSQLExpression`. Then here we can just use whatever v2 expression returned by `PushableExpression`.

Reply:
OK